src/MetaMorpho.sol

```diff
@@ -687,11 +700,11 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
                 revert ErrorsLib.MaxQueueLengthExceeded();
             }
 
-            // Safe "unchecked" cast because withdrawQueue.length <= MAX_QUEUE_LENGTH.
-            marketConfig.withdrawRank = uint64(withdrawQueue.length);
+            marketConfig.enabled = true;
         }
 
         marketConfig.cap = supplyCap;
+        marketConfig.removableAt = 0;
```

Comment:
so setting a cap reset the action of disabling a market

Reply:
Yes otherwise an allocator can freely remove a previously disabled market which was re-enabled

Reply:
it's more a not to my self. I wonder if we should document it or leave as it is

Reply:
I actually moved it to where we set `enabled = false`
So it is always possible to have a market with a non-zero cap and a timelock for it to be removable